File: mili-server/src/main/java/fun/bm/mili/config/modules/function/RedStoneConfig.java
package fun.bm.mili.config.modules.function;

import me.earthme.luminol.config.IConfigModule;
import me.earthme.luminol.config.flags.ConfigClassInfo;
import me.earthme.luminol.config.flags.ConfigInfo;
import me.earthme.luminol.config.flags.TransformedConfig;
import me.earthme.luminol.enums.EnumConfigCategory;

@ConfigClassInfo(category = EnumConfigCategory.FUNCTION, name = "redstone")
public class RedStoneConfig implements IConfigModule {
    @TransformedConfig(name = "shears_rotate", directory = {"misc", "redstone"})
    @TransformedConfig(name = "allow_skip_cooldown", directory = {"misc", "redstone"})
    @ConfigInfo(name = "shears_rotate", comments =
            """
                    允许使用剪刀右键旋转方块。""")
    public static boolean shears = false;
}

File: mili-server/src/main/kotlin/fun/bm/mili/config/modules/function/RedStoneConfig.kt
package `fun`.bm.mili.config.modules.function

import com.electronwill.nightconfig.core.file.CommentedFileConfig
import `fun`.bm.mili.config.modules.ConfigModule
import me.earthme.luminol.config.flags.ConfigClassInfo
import me.earthme.luminol.config.flags.ConfigInfo
import me.earthme.luminol.enums.EnumConfigCategory

@ConfigClassInfo(category = EnumConfigCategory.FUNCTION, name = "redstone-function")
class RedStoneConfig : ConfigModule {
    companion object {
        @ConfigInfo(name = "enabled", comments = "Enable redstone function") @JvmField var enabled = false
        @ConfigInfo(name = "cce", comments = "Shulker box CCE reintroduced") @JvmField var cce = false
        @ConfigInfo(name = "shears", comments = "Enable shears wrench") @JvmField var shears = false
    }
    override fun onLoaded(c: CommentedFileConfig) {}
    override fun onUnloaded(c: CommentedFileConfig) {}
}
